<!-- -*- mode: XML -*- -->
<launch>

	<arg name="control_hz" default="50" />
	<arg name="track_width" default="0.22" />
	<arg name="wheel_radius" default="0.05" />

	<arg name="odom_topic" default="odom" />
	<arg name="cmd_topic" default="cmd_vel" />
	<arg name="l_wheel_topic" default="wheel_left/velocity_controller/command" />
	<arg name="r_wheel_topic" default="wheel_right/velocity_controller/command" />
	<arg name="joint_states_topic" default="joint_states" />

	<arg name="odom_frame_name" default="odom" />
	<arg name="base_frame_name" default="base_footprint" />

	<arg name="use_odom_tf" default="true" />

	<!-- Load joint controller configurations from YAML file to parameter server -->
	<rosparam file="$(find rrcar_control)/config/rrcar_control.yaml" command="load"/>

	<!-- load the controllers -->
	<node name="rrcar_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
		output="screen" ns="rrcar" args="wheel_left/velocity_controller wheel_right/velocity_controller joint_state_controller"/>

	<node name="rrcar_controller" pkg="rrcar_control" type="rrcar_control" ns="rrcar">
		<param name="control_hz" value="$(arg control_hz)"/>
		<param name="track_width" value="$(arg track_width)"/>
		<param name="wheel_radius" value="$(arg wheel_radius)"/>

		<param name="odom_topic" value="$(arg odom_topic)"/>
		<param name="cmd_topic" value="$(arg cmd_topic)"/>
		<param name="l_wheel_topic" value="$(arg l_wheel_topic)"/>
		<param name="r_wheel_topic" value="$(arg r_wheel_topic)"/>
		<param name="joint_states_topic" value="$(arg joint_states_topic)"/>

		<param name="odom_frame_name" value="$(arg odom_frame_name)"/>
		<param name="base_frame_name" value="$(arg base_frame_name)"/>

		<param name="use_odom_tf" type="bool" value="$(arg use_odom_tf)"/>
	</node>

</launch>
